pay-queue/src/pay_queue/external/gcp_auth.py

```diff
@@ -13,28 +13,31 @@
 
 
 def verify_jwt(session):
-    """Verify token is valid."""
-    msg = ''
+    """Verify token is valid and check email claims."""
     try:
         # Get the Cloud Pub/Sub-generated JWT in the "Authorization" header.
-        id_token.verify_oauth2_token(
+        claim = id_token.verify_oauth2_token(
             request.headers.get('Authorization').split()[1],
             Request(session=session),
-            audience=current_app.config.get('PAY_SUB_AUDIENCE')
+            audience=current_app.config.get('PAY_SUB_AUDIENCE') or 'pay-dev'
         )
-    except Exception as e:  # TODO fix
-        msg = f'Invalid token: {e}\n'
-    finally:
-        return msg
+        # Check if the email is verified and matches the configured email
+        required_email = current_app.config.get('VERIFY_PUBSUB_EMAIL') or 'sa-queue@gtksf3-dev.iam.gserviceaccount.com'
```

Comment:
Is this default okay? if this isn't configured in an upper environment it will default to this value that is used for dev?